### ISSUE TITLE
Mouseover values disappear

### ISSUE THREAD

**Issue by AlistairNWard** [tonydisera/gene.iobio#358]
If you load a gene and mouse over the variant ranking table before the coverage has been loaded, the values all get cleared out once the coverage data is loaded. This can be reproduced looking at the demo data set (exome).

**Comment by AlistairNWard** [comment]
<img width="358" alt="screen shot 2016-07-09 at 9 02 13 pm" src="https://cloud.githubusercontent.com/assets/523111/16711108/a142b854-4618-11e6-9b1e-ff09e38cd0df.png">

**Comment by tonydisera** [comment]
Oh, yeah.  Thanks for entering this issue.  I saw it last week as well. :-)

**Comment by tonydisera** [comment]
I can't seem to reproduce this.   There is a possibility that when I reworked the order of variant annotation and coverage (all variants for trio returned first, then coverage calculated), the bug was fixed in the process.  

Would you mind giving this another try to see if the problem has gone away?

**Comment by AlistairNWard** [comment]
Yes, it seems as though this is resolved, however, I am seeing some of the variants only showing info for the proband. Using the demo exome, this screenshot show one such variant where there is only data for the proband, but father has 1x coverage, so has at least one allele.

**Comment by AlistairNWard** [comment]
![untitled](https://cloud.githubusercontent.com/assets/523111/17004127/7cea3b6c-4ea1-11e6-9060-57b64feda016.png)

**Comment by tonydisera** [comment]
In this case below, the allele counts for father is missing.  Looking at the VCF record, the genotype field is blank for the father.  Should a blank genotype be treated as homozygous reference?
<img width="1260" alt="screenshot 2016-07-20 21 25 01" src="https://cloud.githubusercontent.com/assets/8420103/17010413/ef74aa7c-4ec0-11e6-89e8-9fcc239bed37.png">

**Comment by tonydisera** [comment]
Okay, getting back to the relevant comment that you made, yes I see that the father has 1x coverage, but unless there is coverage at the exact position, what displays  is just approximate coverage, constrained by the number of pixels shown.  So for example, the region for RAI1 is 129,980 bp.  But the screen probably only shows about 5000 pixels for the coverage area chart.

Looking at the VCF record, there isn't any genotype info for mother or father.
<img width="1349" alt="screenshot 2016-07-20 21 43 29" src="https://cloud.githubusercontent.com/assets/8420103/17010713/a7e81042-4ec3-11e6-8c5c-559997afd20e.png">

**Comment by AlistairNWard** [comment]
This was laziness on my part! I should have known the cause and I think I have it mentioned in a different issue but can't find it. We should be handling no calls in a way that makes sense and so we could do a number of things:

1. If there is no genotype, we cannot assign one to it, but we should indicate that no call was made. Instead of the 'hom' or 'het' box, we could put 'no call'.

2. If we want to get the coverage at the exact location we can still do that directly from the bam file. e.g. the following command:

samtools view NA19238.bam 17:17644396-17644396 | wc -l

returns a list of all reads that intersect the specific position and then counts them. This returns 3, 0 and 2 for the 3 samples which agrees with the vcf records. So, we could still get the exact coverage for each sample for the mouse over, but we can't assign them to ref or alt without processing the actual read.

In conclusion - I would suggest that where there is a no call, we pull the coverage for the specific site from the bam, and then in the allele count put 'no call' in the box and a different coloured bar showing the allele count for unknown alleles. Make sense?

**Comment by tonydisera** [comment]
Yes, I like this idea.  :-)

**Comment by AlistairNWard** [comment]
#274 is handling the no call issue
